app/shared/buttons/fireGTMButtonClickEvent.js

```diff
@@ -0,0 +1,6 @@
+export const fireGTMButtonClickEvent = (target) => {
+  window.dataLayer?.push({
+    event: 'buttonClick',
+    formId: target.id,
```

Comment:
```
target?.id
```
since this is a global element and we don't want it to fail.
Maybe even a try-catch around it.

Reply:
Using an argument default w/ existential check should avoid any uncaught exceptions here... https://github.com/thegoodparty/gp-webapp/pull/155/commits/c9dabb52deb6ce4bc87260130411ae3841d8ac79